Fabric.Identity.API/scripts/Install-IdPSS-Utilities.psm1

```diff
@@ -64,9 +76,22 @@ function Remove-AzureADClientSecret{
     $encoding = [System.Text.Encoding]::ASCII
     $keys = Get-AzureADApplicationPasswordCredential -ObjectId $objectId
     $filteredKeys = $keys | Where-Object {$null -ne $_.CustomKeyIdentifier -and $encoding.GetString($_.CustomKeyIdentifier) -eq $keyIdentifier}
+
+    $completed = $false
+    [int]$retryCount = 0
     foreach($key in $filteredKeys) {
-        Write-Host "Removing existing password credential named `"$($encoding.GetString($key.CustomKeyIdentifier))`" with id $($key.KeyId)"
-        Remove-AzureADApplicationPasswordCredential -ObjectId $objectId -KeyId $key.KeyId
+        do {
+            Write-Host "Removing existing password credential named `"$($encoding.GetString($key.CustomKeyIdentifier))`" with id $($key.KeyId)"
+            try {
+                Remove-AzureADApplicationPasswordCredential -ObjectId $objectId -KeyId $key.KeyId -ErrorAction 'stop'
+                $completed = $true
+            }
+            catch {
+                Write-DosMessage -Level "Warning" -Message "An error occurred trying to remove the Azure application secret named `"$($encoding.GetString($key.CustomKeyIdentifier))`" with id $($key.KeyId). Retrying.."
+                Start-Sleep 3
+                $retryCount++
```

Comment:
Unless I missed something, it doesn't look like we break out of the do while if we exceed n number of retries...

Reply:
In the try catch line 87 if no error is thrown it continues

Reply:
right, but what if it keeps throwing an exception? I don't see where we short circuit the do while loop in $completed never gets flipped to $true...

Reply:
Oh I see, ya i missed that check